physics/geopotential_test.cpp

```diff
@@ -549,6 +552,249 @@ TEST_F(GeopotentialTest, ThresholdComputation) {
+                                      Length const& r) {
+        Vector<double, ICRS> const down = -direction;
+        return InnerProduct(get_acceleration(geopotential, r), down);
+      };
+  auto const get_latitudinal_acceleration =
+      [&direction, &get_acceleration](Geopotential<ICRS> const& geopotential,
+                                      Length const& r) {
+        Vector<double, ICRS> const celestial_north({0, 0, 1});
+        Vector<double, ICRS> local_north =
+            celestial_north.OrthogonalizationAgainst(direction);
+        return InnerProduct(get_acceleration(geopotential, r), local_north);
+      };
+  auto const get_longitudinal_acceleration =
+      [&direction, &get_acceleration](Geopotential<ICRS> const& geopotential,
+                                      Length const& r) {
+        Vector<double, ICRS> const down = -direction;
+        Bivector<double, ICRS> const north({0, 0, 1});
+        Vector<double, ICRS> local_east = down * north;
+        return InnerProduct(get_acceleration(geopotential, r), local_east);
+      };
+
+  // Above the  outer threshold for J2.
+  EXPECT_THAT(
+      get_acceleration(earth_geopotential, 5'000'000 * Kilo(Metre)).Norm(),
+      Eq(0 / Pow<2>(Metre)));
+
+  {
+    // Inspect the J2 sigmoid.
+    Length const s0 = earth_geopotential.degree_damping()[2].inner_threshold();
+    Length const s1 = earth_geopotential.degree_damping()[2].outer_threshold();
+    EXPECT_THAT(s0, IsNear(1'500'000 * Kilo(Metre)));
+
+    // The radial component grows beyond the undamped one.  We check the ratio
+    // at the arithmetic mean of the thresholds, and at its maximum.
+    EXPECT_THAT(
+        get_radial_acceleration(earth_geopotential, (s0 + s1) / 2) /
+            get_radial_acceleration(*geopotential_j2, (s0 + s1) / 2),
+        AlmostEquals(1, 0));
+    EXPECT_THAT(
+        get_radial_acceleration(earth_geopotential, 2 * s0 * s1 / (s0 + s1)) /
+            get_radial_acceleration(*geopotential_j2, 2 * s0 * s1 / (s0 + s1)),
+        AlmostEquals(1.125, 2));
+
+    // The latitudinal component remains below the undamped one (it is simply
+    // multiplied by σ, instead of involving σ′).
+    EXPECT_THAT(
+        get_latitudinal_acceleration(earth_geopotential, (s0 + s1) / 2) /
+            get_latitudinal_acceleration(*geopotential_j2, (s0 + s1) / 2),
+        AlmostEquals(0.5, 2));
+  }
+
+  // Below the inner threshold for J2, but still above all other outer
+  // thresholds.
+  EXPECT_THAT(earth_geopotential.degree_damping()[2].inner_threshold(),
+              Gt(1'000'000 * Kilo(Metre)));
+  EXPECT_THAT(earth_geopotential.tesseral_damping().outer_threshold(),
+              Lt(1'000'000 * Kilo(Metre)));
+  EXPECT_THAT(earth_geopotential.degree_damping()[3].outer_threshold(),
+              Lt(1'000'000 * Kilo(Metre)));
+  EXPECT_THAT(get_acceleration(earth_geopotential, 1'000'000 * Kilo(Metre)),
+              Eq(get_acceleration(*geopotential_j2, 1'000'000 * Kilo(Metre))));
```

Comment:
Use `AlmostEquals` with a tolerance.

Reply:
No, this is equality because the same path is taken, not because different calculations give the same result.